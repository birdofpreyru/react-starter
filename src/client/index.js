import { client, redux } from 'topcoder-react-utils';

client({
  applicationModulePath: require.resolve('../shared'),
  /* eslint-disable global-require */
  getApplication: () => require('../shared').default,
  /* eslint-enable global-require */
  moduleHot: module.hot,
  storeFactory: async initialState => redux.storeFactory({
    /* eslint-disable global-require */
    getReducerFactory: () => require('reducers').factory,
    /* eslint-enable global-require */
    initialState,
    moduleHot: module.hot,
    reducerFactoryModulePath: require.resolve('reducers'),
  }),
});
