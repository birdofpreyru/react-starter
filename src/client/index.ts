// Note: We don't use "import" statements here, as they would require
// to check import.meta.webpackHot below for isHmrEnabled check,
// and that's break Jest tests, which do not support ES modules yet.

const { client } = require('@dr.pogodin/react-utils');

const Application = require('../shared').default;

const isHmrEnabled = !!(module as any).hot;

client(Application, {
  // With HMR hydration may throw hydration errors in subsequent re-renders.
  dontHydrate: isHmrEnabled,
});
