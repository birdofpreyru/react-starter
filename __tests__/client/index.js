jest.mock('shared', () => ({
  default: 'APPLICATION',
}));

jest.mock('reducers', () => ({
  factory: () => () => ({}),
}));

jest.mock('topcoder-react-utils', () => {
  const TRU = require.requireActual('topcoder-react-utils');
  return {
    ...TRU,
    client: jest.fn(),
  };
});

test('Passes basic testing', () => {
  require('client');
  const { client } = require('topcoder-react-utils');

  const args = client.mock.calls[0];
  expect(args).toMatchSnapshot();

  expect(args[0].getApplication()).toBe('APPLICATION');

  expect(args[0].storeFactory()).toBeInstanceOf(Promise);
});
