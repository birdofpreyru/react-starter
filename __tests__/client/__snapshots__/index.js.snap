// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes basic testing 1`] = `
Array [
  Object {
    "applicationModulePath": "/home/birdofprey/Documents/topcoder/Topcoder/topcoder-react-starter/src/shared/index.jsx",
    "getApplication": [Function],
    "moduleHot": undefined,
    "storeFactory": [Function],
  },
]
`;
